### PR TITLE
Home page

### PR DESCRIPTION
**Pull request by waliens**
First version for the (connected user) home page

### CODE REVIEW

**Comment by waliens** [comment]
This requires the latest api version (for user stats and latest activities requests) !

File: boardgame-ui/src/components/activities/ActivityBox.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                <br/>
+                <p class="activity-datetime">{{activity.datetime}}</p>
+            </div>
+        </article>
+        <article class="media" v-if="activity.type === 'library/add'">
+            <div class="media-left">
+                <figure class="image is-64x64"><img class="is-rounded" :src=activity.board_game.thumbnail /></figure>
+            </div>
+            <div class="media-content">
+                {{$t("activity.library.added")}} <strong>{{activity.board_game.name}}</strong> {{$t("activity.library.library")}}.
```
Comment (elodieburtin) [inline boardgame-ui/src/components/activities/ActivityBox.vue]: Better to use named formatting  or list formatting (https://kazupon.github.io/vue-i18n/guide/formatting.html#named-formatting) offered by i18n rather than inserting variables this way.
```suggestion
                {{$t("activity.library.added_game", {gameName: activity.board_game.name})}}.
```
See https://developer.mozilla.org/en-US/docs/Mozilla/Localization/Localization_content_best_practices#Word_Order

If you want a particular style, you can also use https://kazupon.github.io/vue-i18n/guide/interpolation.html#basic-usage
```suggestion
                <i18n path="activity.library.added_game" tag="p">
                  <strong place="gameName">{{ activity.board_game.name }}</strong>
                </i18n>
```

File: boardgame-ui/src/locales/translations.csv [hunk 2]
```diff
@@ -1,4 +1,19 @@
...
+home.stats.played,Games played,
+home.stats.attended,Events attended,
+home.stats.owned,Games owned,
+home.stats.most_played,Most played,
+home.stats.time_played,Time played,
+home.stats.minutes,min.,
+home.stats.times,times,
+activity.event.join,You joined,
+activity.game.played,You played to,
+activity.library.added,You added,
```
Comment (elodieburtin) [inline boardgame-ui/src/locales/translations.csv]:
```suggestion
activity.library.added_game,You added {gameName} to your library,
```

File: boardgame-ui/src/router.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const guestOnly = async (to, from, next) => {
     next();
     return;
   }
-  next({name: 'home'});
+  next({name: 'login'});
```
Comment (elodieburtin) [inline boardgame-ui/src/router.js]: Why this change?

File: boardgame-ui/src/pages/HomePage.vue [hunk 4]
```diff
@@ -1,3 +1,129 @@
...
+  components: {
+    ActivityBox
+  },
+  computed: {
+    currentUser() {
+      return this.$store.state.currentUser;
+    }
+  },
+  async created() {
+    if (!this.currentUser) {
```
Comment (elodieburtin) [inline boardgame-ui/src/pages/HomePage.vue]: As we discussed, if this line is executed before the store is initialized, the data is not fetched, and the page is never populated.
Several solutions to avoid this issue:
1. Create a new component AuthenticatedHomePage, include in it all data and template related to current user (line 19 to 62), display it only if there is a current user (v-if="currentUser") and fetch data in the created hook of this new component
2. In App.vue, prevent the creation of <router-view> component until the store has been initialized
3. Create a watcher for currentUser, and reload data if its value is changed

I think the second is the easiest, but I let you decide, and don't hesitate to ask if you need help! :-)

File: boardgame-ui/src/components/activities/ActivityBox.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
+<template>
+    <div v-if=activity class="activity-container">
+        <article class="media" v-if="activity.type === 'event/join'">
+            <div class="media-left calendar-media-left">
+                <figure class="is-64x64"><span><i class="fa-3x far fa-calendar-alt"></i></span></figure>
+            </div>
+            <div class="media-content">
+                {{$t("activity.event.join")}} <strong>{{activity.event.name}}</strong>.
+                <br/>
+                <p class="activity-datetime">{{activity.datetime}}</p>
```
Comment (elodieburtin) [inline boardgame-ui/src/components/activities/ActivityBox.vue]: We could use moment.js for more user-friendly date display (but not needed for now I think ^^)

File: boardgame-ui/src/pages/HomePage.vue (hunk 4 above)
Comment (waliens) [reply boardgame-ui/src/pages/HomePage.vue]: Actually, there would be very few common content between the authenticated home page and the other. I think it would be better to have two different pages. I was not sure how to update the router (the change of "home" to "login" redirect was a temporary workaround in this direction), especially because the path would be the same (i.e. '/'). I was not sure if it was possible to have several pages with the same path.

File: boardgame-ui/src/components/activities/ActivityBox.vue (hunk 5 above)
Comment (waliens) [reply boardgame-ui/src/components/activities/ActivityBox.vue]: Indeed, but I think we should do more: like develop few functions for displaying dates throughout the application. We need this because the API returns all datetimes in UTC so we need to convert them to the user timezone. I could work on that if you agree on the idea.

File: boardgame-ui/src/router.js (hunk 3 above)
Comment (waliens) [reply boardgame-ui/src/router.js]: see comment on currentUser

File: boardgame-ui/src/pages/HomePage.vue (hunk 4 above)
Comment (elodieburtin) [reply boardgame-ui/src/pages/HomePage.vue]: That's why I think it would make sense to have one component responsible for the home page of the guest (let us call it GuestHomePage), and one responsible for the home page of an authenticated user (UserHomePage).
Then, the template of HomePage.vue would simply be:
```
<template>
  <user-home-page v-if="currentUser"></user-home-page>
  <guest-home-page v-else></guest-home-page>
</template>
```
I don't know if there is another way to handle this in vue-router, I can investigate :-)

File: boardgame-ui/src/locales/translations.csv (hunk 2 above)
Comment (waliens) [reply boardgame-ui/src/locales/translations.csv]: Great, i will correct that !

File: boardgame-ui/src/components/activities/ActivityBox.vue (hunk 5 above)
Comment (elodieburtin) [reply boardgame-ui/src/components/activities/ActivityBox.vue]: Don't you think http://momentjs.com/ and http://momentjs.com/timezone/ cover our needs?

File: boardgame-ui/src/router.js (hunk 3 above)
Comment (elodieburtin) [reply boardgame-ui/src/router.js]: OK, but better to remove it (it causes my browser to crash when I try to access the login page while authenticated, since it creates an infinite redirection loop)

File: boardgame-ui/src/pages/HomePage.vue (hunk 4 above)
Comment (waliens) [reply boardgame-ui/src/pages/HomePage.vue]: Let's go for this solution then 👍

File: boardgame-ui/src/components/activities/ActivityBox.vue (hunk 5 above)
Comment (waliens) [reply boardgame-ui/src/components/activities/ActivityBox.vue]: We should use this indeed, but I don't want for instance the format string to be repeated everywhere. I think it would be great to have a small function on top of those.

Comment (waliens) [reply boardgame-ui/src/components/activities/ActivityBox.vue]: I created a new component bgc-datetime to encapsulate datetime formatting !
